assets/src/components/v2/pre_fare_single_screen_alert.tsx

```diff
@@ -235,10 +243,29 @@ const remedySection = (
 );
 
 const mapSection = (disruptionDiagram?: DisruptionDiagramData) => {
```

Comment:
Little update after I did some more reading. My tl;dr definitely simplifies things a bit; it's more like "Function that returns JSX and requires lifecycle/state == React component", but sticking with the simpler definition won't cause problems and will make the component tree in react dev tools much more informative.

**Component `A` calls component `B` as a function** ->
- The JSX `B` returns gets treated as part of `A`'s overall return.
- `B` is evaluated immediately if `A` re-renders.
- `B` is invisible to React and does not get state or a lifecycle.

**Component `A` invokes component `B` as a component** ->
- The JSX `B` returns is nested within `B` in the render tree.
- `B` is evaluated only if needed (lazy evaluation).
- `B` is visible to React and has a lifecycle / can keep state / can use hooks.

Reply:
`standardLayout` and `fallbackLayout`, at minimum, also need to be converted to components since they use hooks. Ideally all of the functions that return JSX in this file would be converted to components.

I'd recommend using JSX to invoke all of the components that can be returned by that `layoutRenderer` function. If you like, you could also flatten it out with an IIFE or a `let`:

```tsx
// IIFE style:
// Define the 0-argument anonymous function, and immediately call it.
// A workaround for `switch` being a statement instead of a value-returning expression.
const layout = (() => {
  switch (true) {
    case effect === "delay":
      return <FallbackLayout issue={issue} remedy={remedy} effect={effect} routes={routes} />;
    ...
  }
})();

// OR

// `let` style:
let layout;
switch (true) {
  case effect === "delay":
    layout = <FallbackLayout issue={issue} remedy={remedy} effect={effect} routes={routes} />;
  ...
}

// ---

// Down below in the main render:
...
<div className="alert-card__body">{layout}</div>
...
```

Reply:
Yeah probably a good idea to tackle this now. I was going to say it's out of scope for this, but it wasn't that difficult to do.